tox.ini

```diff
@@ -23,8 +22,9 @@ commands =
     pre-commit install -f --install-hooks
 
 [testenv:pre-commit]
+basepython = python3.7
 deps = pre-commit >= 1.16.1
-commands = pre-commit {posargs}
+commands = pre-commit run --all-files
```

Comment:
Huh, was there a reason for this change?

FWIW, `{posargs}` allows the invocation like:

```
tox -e pre-commit -- run --all-files
```

which is somewhat handy, if you don't want (or have trouble with) installing pre-commit in your current environment.

Reply:
My understanding is that `tox -e pre-commit -- run --all-files` is equivalent to `tox -e pre-commit` with the above contents of the `pre-commit` test environment (in regards to not being able to install `pre-commit` in the current environment). I switched the `pre-commit {posargs} to `pre-commit run --all-files`, because once I pulled `pre-commit` into its own env, it wasn't running on all files with `{posargs}` in the command.

Reply:
Weird. Ok. I guess we just won't support use cases like:

```
tox -e pre-commit -- run
```

for now then.